cli/flox-rust-sdk/src/models/manifest/composite/shallow.rs

```diff
@@ -0,0 +1,499 @@
+            if cfg_hp.working_dir.is_some() {
+                prop_assert_eq!(merged.working_dir, cfg_hp.working_dir);
+            } else {
+                prop_assert_eq!(merged.working_dir, cfg_lp.working_dir);
+            }
+        }
+
+        #[test]
+        fn containerize_cfg_deep_merges_labels(
+            cfg_lp in any::<ManifestContainerizeConfig>(),
+            cfg_hp in any::<ManifestContainerizeConfig>(),
+        ) {
+            let merged = deep_merge_optional_containerize_config(Some(&cfg_lp), Some(&cfg_hp)).unwrap();
+            match (cfg_lp.labels, cfg_hp.labels) {
+                (None, None) => prop_assert!(merged.labels.is_none()),
+                (Some(lp), None) => {
+                    prop_assert_eq!(merged.labels, Some(lp));
+                }
+                (None, Some(hp)) => {
+                    prop_assert_eq!(merged.labels, Some(hp));
+                }
+                (Some(lp), Some(hp)) => {
+                    prop_assert!(merged.labels.is_some());
+                    let merged_labels = merged.labels.unwrap();
+                    for key in merged_labels.keys() {
+                        if hp.contains_key(key) {
+                            prop_assert_eq!(merged_labels.get(key), hp.get(key));
+                        } else {
+                            prop_assert_eq!(merged_labels.get(key), lp.get(key));
+                        }
+                    }
+                }
+            }
+        }
+
+        #[test]
+        fn containerize_cfg_shallow_merges_stop_signal(
+            cfg_lp in any::<ManifestContainerizeConfig>(),
+            cfg_hp in any::<ManifestContainerizeConfig>(),
+        ) {
+            let merged = deep_merge_optional_containerize_config(Some(&cfg_lp), Some(&cfg_hp)).unwrap();
+            if cfg_hp.stop_signal.is_some() {
+                prop_assert_eq!(merged.stop_signal, cfg_hp.stop_signal);
+            } else {
+                prop_assert_eq!(merged.stop_signal, cfg_lp.stop_signal);
+            }
+        }
+
+        #[test]
+        fn containerize_deep_merges_config(
+            cfg_lp in any::<ManifestContainerizeConfig>(),
+            cfg_hp in any::<ManifestContainerizeConfig>()
+        ) {
+            let cont_lp = ManifestContainerize { config: Some(cfg_lp.clone())};
+            let cont_hp = ManifestContainerize { config: Some(cfg_hp.clone())};
+            let maybe_merged = ShallowMerger::merge_containerize(Some(&cont_lp), Some(&cont_hp)).unwrap();
+            prop_assert!(maybe_merged.is_some()); // They were both Some(_) to start out
+            let merged_cont = maybe_merged.unwrap();
+            prop_assert!(merged_cont.config.is_some());
+            let merged_cfg = merged_cont.config.unwrap();
+            // This is essentially checking that the deep merge happens at all.
+            // The details/correctness of the deep merge are verified by the
+            // more focused tests above.
```

Comment:
this might be better suited as the tests doc comment

Reply:
Fixed